pulsar-broker/src/main/java/org/apache/pulsar/compaction/TwoPhaseCompactor.java

```diff
@@ -120,9 +123,6 @@ private void phaseOneLoop(RawReader reader,
             return;
         }
         CompletableFuture<RawMessage> future = reader.readNextAsync();
-        FutureUtil.addTimeoutHandling(future,
```

Comment:
The scheduled task will be canceled when the `future` is complete, and the second scheduled task will only be created after the first `future` is completed, so it will not cause thousands of scheduled tasks.

Reply:
Originally i am worried that the frequent tasks schedule and cancellation will impact the performance, but it seems not.